pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -1058,17 +1058,17 @@ func (configgen *ConfigGeneratorImpl) buildHTTPProxy(node *model.Proxy,
 	push *model.PushContext, proxyInstances []*model.ServiceInstance) *xdsapi.Listener {
 	httpProxyPort := push.Mesh.ProxyHttpPort
 	noneMode := node.GetInterceptionMode() == model.InterceptionNone
-	_, actualLocalHostAddress := getActualWildcardAndLocalHost(node)
-
-	if httpProxyPort == 0 && noneMode { // make sure http proxy is enabled for 'none' interception.
-		httpProxyPort = int32(features.DefaultPortHTTPProxy)
-	}
-	// enable HTTP PROXY port if necessary; this will add an RDS route for this port
 	if httpProxyPort == 0 {
-		return nil
+		// make sure http proxy is enabled for 'none' interception.
+		if noneMode {
+			httpProxyPort = int32(features.DefaultPortHTTPProxy)
+		} else {
+			return nil
```

Comment:
you are changing logic here. it was not in else before

Reply:
nit changing, they both return when the `httpProxyPort == 0`